src/systems/equation_systems.C

```diff
@@ -218,6 +200,7 @@ bool EquationSystems::reinit_solutions ()
 
         sys.prolong_vectors();
       }
+    //FIXME: why have this if we set it to 'true' anyway?
```

Comment:
IIRC, once upon a time this was *supposed* to be a part of the forwards compatibility path towards requiring users to not handle the `refine_and_coarsen_elements()` calls themselves, but to merely flag elements and allow the EquationSystems to handle all mesh refinement and coarsening as it was doing projections.  I never figured out how to properly get that to work without breaking old codes.  It might be worth revisiting, though; the current system makes AMR more expensive than it should be.